src/qibo/quantum_info/entropies.py

```diff
@@ -56,17 +57,19 @@ def shannon_entropy(prob_dist, base: float = 2, backend=None):
 
     total_sum = backend.np.sum(prob_dist)
 
-    if np.abs(total_sum - 1.0) > PRECISION_TOL:
+    if np.abs(backend.to_numpy(total_sum) - 1.0) > PRECISION_TOL:
```

Comment:
I'm quite sure that this would work since `total_sum` is just a `foat`.
```suggestion
    if np.abs(float(total_sum) - 1.0) > PRECISION_TOL:
```
This is valid everywhere else this type of structure appears